src/System.Text.Primitives/System/Text/Parsing/Unsigned.cs

```diff
@@ -0,0 +1,301 @@
+// in Unsigned.tt.
+
+namespace System.Text
+{
+    public static partial class PrimitiveParser
+    {
+        public static bool TryParseByte(ReadOnlySpan<byte> text, out byte value, out int bytesConsumed, EncodingData encoding = default(EncodingData), TextFormat format = default(TextFormat))
+        {
+            if (format.IsHexadecimal)
+            {
+                if (encoding.IsInvariantUtf8)
+                {
+                    return InvariantUtf8.Hex.TryParseByte(text, out value, out bytesConsumed);
+                }
+                else if (encoding.IsInvariantUtf16)
+                {
+                    ReadOnlySpan<char> utf16Text = text.Cast<byte, char>();
+                    int charsConsumed;
+                    bool result = InvariantUtf16.Hex.TryParseByte(utf16Text, out value, out charsConsumed);
+                    bytesConsumed = charsConsumed * sizeof(char);
+                    return result;
+                }
+                else
+                {
+                    // TODO: EncodingData.InvariantUtf8 and EncodingData.InvariantUtf16 don't work for parsing right now
+                    // because their parsing trees are undefined. A custom EncodingData with a manually-created parsing
+                    // tree must be used instead. Default to InvariantUtf8 parsing for now, but after autogeneration of
+                    // parsing trees is implemented, this should return false.
+                    return InvariantUtf8.Hex.TryParseByte(text, out value, out bytesConsumed);
+                }
+            }
+
+            uint nextSymbol;
+            int thisSymbolConsumed;
+            if (!encoding.TryParseSymbol(text, out nextSymbol, out thisSymbolConsumed))
+            {
+                value = default(byte);
+                bytesConsumed = 0;
+                return false;
+            }
+
+            if (nextSymbol > 9)
+            {
+                value = default(byte);
+                bytesConsumed = 0;
+                return false;
+            }
+
+            uint parsedValue = nextSymbol;
+            int index = thisSymbolConsumed;
+
+            while (index < text.Length)
+            {
+                bool success = encoding.TryParseSymbol(text.Slice(index), out nextSymbol, out thisSymbolConsumed);
+                if (!success || nextSymbol > 9)
+                {
+                    bytesConsumed = index;
+                    value = (byte) parsedValue;
+                    return true;
+                }
+
+                // If parsedValue > (byte.MaxValue / 10), any more appended digits will cause overflow.
+                // if parsedValue == (byte.MaxValue / 10), any nextDigit greater than 5 implies overflow.
+                if (parsedValue > byte.MaxValue / 10 || (parsedValue == byte.MaxValue / 10 && nextSymbol > 5))
+                {
+                    bytesConsumed = 0;
```

Comment:
Design questions: should we return the current index when parsing fails? This would allow programs (e.g. editor) to show to the user the exact location where parsers failed.